portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -87,10 +94,16 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     String username = request.getParameter("username");
     String comment = request.getParameter("comment");
 
+    BlobstoreService blobstoreService = BlobstoreServiceFactory.getBlobstoreService();
+    Map<String, List<BlobKey>> blobs = blobstoreService.getUploads(request);
+    List<BlobKey> blobKeys = blobs.get("image");
```

Comment:
For field types, you should generally use the immutable type (such as ImmutableList) instead of the general collection interface type (such as List and Map). This communicates within the class the semantic guarantees of immutable types, which is almost always very useful information. When it's a general collection type, future editors of the class may be under the impression that they can mutate the field, leading to failures at runtime.

go/prefer-immutable-collections